### PR TITLE
promtailclient reconciler

### PR DESCRIPTION
**Pull request by hervenicol**
Towards https://github.com/giantswarm/giantswarm/issues/27146

Add promtailclient reconciler

Currently missing cluster name detection for vintageMC, only works on `gauss`.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -60,12 +61,14 @@ func init() {
...
 func main() {
 	var enableLeaderElection bool
+	var installName string
 	var metricsAddr string
 	var probeAddr string
 	var vintageMode bool
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&installName, "install-name", "unknown", "Name of the installation")
```
Comment (QuentinBisson) [inline main.go:71]:
```suggestion
	flag.StringVar(&installationName, "install-name", "unknown", "Name of the installation")
```

Comment (hervenicol) [reply main.go]: Done :heavy_check_mark: 
BTW: that's nice of you to do suggestions, but when they require updates in other parts of the code that looks like a trap :rofl:

Comment (QuentinBisson) [reply main.go]: True but I'd rather not do suggestions all around :D

File: main.go [hunk 2]
```diff
@@ -123,6 +126,13 @@ func main() {
 		Client: mgr.GetClient(),
 	}
 
+	promtailClient := promtailclient.Reconciler{
+		Client: mgr.GetClient(),
+	}
+
+	var loggingReconcilerOptions = make(map[string]string)
+	loggingReconcilerOptions["installName"] = installationName
```
Comment (QuentinBisson) [inline main.go:134]:
```suggestion
	loggingReconcilerOptions["installationName"] = installationName
```

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**